Initial documentation review: getting-started (#303)
egil/bUnit: docs/site/docs/providing-input/passing-parameters-to-components.md

## Regular Parameters

A regular parameter is one that is declared using the `[Parameter]` attribute. The following subsections will cover both _non_-Blazor type parameters, e.g. `int` and `List<string>`, and the special Blazor types like `EventCallback` and `RenderFragment`.

### Non-Blazor Type Parameters

> Let's look at an example of passing parameters that takes types which or [__AP: should 'or' be 'are'?__] _not_ special to Blazor, i.e.:

Reviewer: ```suggestion
Let's look at an example of passing parameters that takes types which are _not_ special to Blazor, i.e.:
```

## Rendering a Component Under Test Inside Other Components

It is possible to nest a component under tests inside other components, if that is required to test it. For example, to nest the `<HelloWorld>` component inside the `<Wrapper>` component do the following:

# [C# test code](#tab/csharp)

[!code-csharp[](../../../samples/tests/xunit/NestedComponentTest.cs#L16-L28)]

> These examples do the same thing, i.e. render the `<HelloWorld>` component inside the `<Wrapper>` component. What is special in both cases is the use of the `FindComponent<HelloWorld>()`, which returns a `IRenderedComponent<HelloWorld>`, which gives access to only the `<HelloWorld>` components part of the render tree, and the `<HelloWorld>` components instance.[__AP: The last clause is rather ambiguous. Is it that the IRenderedComponent\<HelloWorld\> gives access to two things - the \<HelloWorld\> components part of the render tree and the \<HelloWorld\> components instance - or that something (\<Wrapper\>?) returns (1) the IRenderedComponent\<HelloWorld\> and (2) the \<HelloWorld\> components instance?__]

Reviewer: ```suggestion
These examples do the same thing, i.e. render the `<HelloWorld>` component inside the `<Wrapper>` component. What is special in both cases is the use of the `FindComponent<HelloWorld>()` that returns a `IRenderedComponent<HelloWorld>`. This is needed because the `RenderComponent<Wrapper>` method call returns an `IRenderedComponent<Wrapper>` instance, that provides access to the instance of the `<Wrapper>` component, but not the `<HelloWorld>`-component instance.
```

> AP: The last clause is rather ambiguous. Is it that the IRenderedComponent\<HelloWorld\> gives access to two things - the \<HelloWorld\> components part of the render tree and the \<HelloWorld\> components instance - or that something (\<Wrapper\>?) returns (1) the IRenderedComponent\<HelloWorld\> and (2) the \<HelloWorld\> components instance?

Is this more clear?

# [Razor test code](#tab/razor)

[!code-cshtml[](../../../samples/tests/razor/NestedComponentTest.razor#L3-)]

This is just regular Blazor child content parameter passing, where one component is rendered inside another, i.e. the `<HelloWorld>` component inside the `<Wrapper>` component. 

> The special thing in this case is that the `GetComponentUnderTest<HelloWorld>()` method specifies the `<HelloWorld>` component as its target instead of the outer `<Wrapper>` component. This returns a `IRenderedComponent<HelloWorld>`, which gives access to only the `<HelloWorld>` components part of the render tree, and the `<HelloWorld>` components instance. [__AP: The last clause is rather ambiguous. Is it that the IRenderedComponent\<HelloWorld\> gives access to two things - the \<HelloWorld\> components part of the render tree and the \<HelloWorld\> components instance - or that something (\<Wrapper\>?) returns (1) the IRenderedComponent\<HelloWorld\> and (2) the \<HelloWorld\> components instance?__]

Reviewer: ```suggestion
The special thing in this case is that the `GetComponentUnderTest<HelloWorld>()` method specifies the `<HelloWorld>` component as its target instead of the outer `<Wrapper>` component. This returns a `IRenderedComponent<HelloWorld>`, the child of the `<Wrapper>` component, making it the component under test.
```

> AP: The last clause is rather ambiguous. Is it that the IRenderedComponent\<HelloWorld\> gives access to two things - the \<HelloWorld\> components part of the render tree and the \<HelloWorld\> components instance - or that something (\<Wrapper\>?) returns (1) the IRenderedComponent\<HelloWorld\> and (2) the \<HelloWorld\> components instance?

Is this more clear?
